Custom base path

jesperfj opened the pull request:
This PR proposes a design for addressing #86. Feedback welcome. The idea is to allow setting a custom base path in `ForceApi`. It is only applicable to the "raw" get/post/put/delete/patch methods and will not affect any other methods in the class. The purpose is to allow this library to be used with Apex REST and maybe other API endpoints outside of the standard `/services/data/<version>` base path.

No tests have been written against this yet. The code compiles and the standard tests pass.

README has details for how this will work. But here it is again:

    ForceApi api = new ForceApi(myConfig,mySession);
    api.withBasePath("").get("/services/apexrest/myApexClass");

@ModeratelyComfortableChair let me know what you think.

ModeratelyComfortableChair reviewed (approved):
Apologies for the late response. This works perfectly. I use the following
ForceApi api = new ForceApi(myConfig);
api = api.withBasePath("");
Input input = new Input();
ResourceRepresentation rr = api.post("/services/apexrest/MyApexClass", input);

jesperfj commented:
I am wondering if I am overcomplicating it. How about this instead:

api.rootPath().get("/services/apexrest/myApexClass");

The idea here is that probably `withBasePath("")` will be the by far most common usage and `rootPath()` supports all use cases. You just have to construct the full path yourself in the get/post/put/patch/delete method call.

jesperfj commented:
I had to force push because I rebased this branch on top of some master changes. the rootPath() approach is now implemented and there is a very simple test.
